tao_compiler/mlir/xla/ral/context/common_context_impl_mkldnn.h

```diff
@@ -314,6 +404,124 @@ bool parseConvParams(ExecutionContext* ctx, MemRefType<Tinput, N> input,
+  // 2 * KH * KW * KIC - 1
+  int64_t flops_per_output =
+      2 * static_cast<int64_t>(params.weight.get_nelems()) / OC - 1;
+  // #outputs * flops_per_output
+  int64_t gflops =
+      static_cast<int64_t>(params.dst.get_nelems()) * flops_per_output;
+
+  std::ostringstream sout;
+  sout << message << ":\n";
+  sout << "  input logical NCHW shape:\n\t";
+  for (const auto& d : src_dims) {
+    sout << d << " ";
+  }
+  sout << "\n  kernel logical OIHW shape:\n\t";
+  for (const auto& d : kernel_dims) {
+    sout << d << " ";
+  }
+  sout << "\n  output logical NCHW shape:\n\t";
+  for (const auto& d : dst_dims) {
+    sout << d << " ";
+  }
+  sout << "\n  strides:\n\t";
+  for (size_t i = 0; i < params.strides.size(); ++i) {
+    sout << params.strides[i] << " ";
+  }
+  sout << "\n  dilates:\n\t";
+  for (size_t i = 0; i < params.dilates.size(); ++i) {
+    sout << params.dilates[i] << " ";
+  }
+  sout << "\n  paddings_l:\n\t";
+  for (size_t i = 0; i < params.padding_l.size(); ++i) {
+    sout << params.padding_l[i] << " ";
+  }
+  sout << "\n  paddings_r:\n\t";
+  for (size_t i = 0; i < params.padding_r.size(); ++i) {
+    sout << params.padding_r[i] << " ";
+  }
+  TAO_VLOG(0) << sout.str() << "\n roofline:\n"
+              << "\tMath Ops = " << gflops << "\n"
+              << "\tBytes = " << bytes << "\n"
+              << "\tBandwidth = " << double(bytes) / double(nanosec) << " GB\n"
+              << "\tGFLOPS = " << double(gflops) / double(nanosec) << "\n";
+}
+
+#if defined(TAO_AARCH64)
+struct AclDepthwiseConvInfo {
+  arm_compute::Tensor src;
+  arm_compute::Tensor weights;
+  arm_compute::Tensor dst;
+  arm_compute::NEDepthwiseConvolutionLayer depthwise_conv;
+};
+
+template <typename TKey, typename TValue>
+using ACLDepthwiseConvMap =
+    std::unordered_map<TKey, TValue, ConvParamsKeyHasher>;
+using ACLDepthwiseCache = ideep::utils::lru_cache<
+    ConvParamsKey, std::shared_ptr<AclDepthwiseConvInfo>, ACLDepthwiseConvMap>;
+
+struct ACLDepthwiseConvState : public Context::Resource {
+  std::mutex mu;
+  ACLDepthwiseCache cache{getWeightPrePackingCacheCapacity()};
+
+  inline std::shared_ptr<AclDepthwiseConvInfo> getOrCreate(
+      const ConvParamsKey& key,
+      std::function<std::shared_ptr<AclDepthwiseConvInfo>()> creator) {
+    std::lock_guard<std::mutex> l(mu);
+    auto it = cache.find(key);
+    if (it == cache.end()) {
+      it = cache.insert(std::make_pair(key, creator())).first;
+    }
+    return it->second;
+  }
+};
+
+struct AclConvInfo {
+  arm_compute::Tensor src;
+  arm_compute::Tensor weights;
+  arm_compute::Tensor dst;
+  arm_compute::NEConvolutionLayer conv;
+};
+
+template <typename TKey, typename TValue>
+using ACLConvMap = std::unordered_map<TKey, TValue, ConvParamsKeyHasher>;
+using ACLConvCache =
+    ideep::utils::lru_cache<ConvParamsKey, std::shared_ptr<AclConvInfo>,
+                            ACLConvMap>;
+
+struct ACLConvState : public Context::Resource {
```

Comment:
`ACLDepthwiseConvState` and `ACLConvState` are similar. I think they can merge into one class definition with template?

Reply:
done